### ISSUE TITLE
Change object format for DzVertexMap object

### ISSUE THREAD

**Issue by sethzenz** [cms-analysis/flashgg#57]
TLDR: we should get rid of maps we put in the event.  I will take care of this but let me know if you need a map and I can help you replace it.  More details below if you're curious.

I have been advised by CMSSW memory management experts that maps can create issues or be unstable, and that `map<T,vector<U> >` is especially bad. [Here T=`Ptr<vertex>` and U=`PackedCandidate`.]  The use of this construct for a one-to-many map can be replaced with some clever handling of `vector<pair<T,U> >`, in which all the U's corresponding to a given T are in sequence and you rely on that to look at only the right part of the vector.

Also, for reference, a one-to-one `map<T,U>` should simply be replaced with a vector<U> for which you rely on the fact that the new collection is in the same order as the old. I had significant trouble creating the map anyway, because CMSSW implicitly requires a `vector<T>` dictionary for a `map<T,U>` even if you don't think you're using `vector<T>`.  I eventually got rid of the map (in code I will do a pull request for later this morning) and just used the `vector<U>`.

**Comment by sethzenz** [comment]
std::unordered_multimap<T, U> now seems to me to be a more intuitive choice.
